lib/mongoid/matchable/default.rb

```diff
@@ -69,6 +69,27 @@ def determine(value, operator)
           attr ? attr.send(operator, first(value)) : false
         }
       end
+
+      # Convenience method for checking _matches? on a Document or a Hash.
+      #
+      # @example Does the document of type Document or Hash match?
+      #   recursive_matches?(default, "a", 1)
+      #   recursive_matches?(:a => 1, "b", 2)
+      #
+      # @param [ Document, Hash ] document The object of type Document or Hash to call _matches? on
+      # @param [ String ] key The key
+      # @param [ Object ] value The value to check if it matches.
+      #
+      # @return [ true, false ] True if matches, false if not.
+      #
+      # @since 7.0.4
```

Comment:
The since versions should be 7.0.5 since 7.0.4 had already been released.

Reply:
fixed